File: 617-merge-two-binary-trees/617-merge-two-binary-trees.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode mergeTrees(TreeNode root1, TreeNode root2) {
        
        if(root1 == null && root2 == null)return null;
        
        Integer new_val = 0;
        
        if(root1 != null) new_val += root1.val;
        if(root2 != null) new_val += root2.val;
        
        TreeNode new_node = new TreeNode(new_val);
        
        new_node.left = mergeTrees( (root1!=null ? root1.left : null ), (root2!=null ? root2.left : null )  );
        
        new_node.right = mergeTrees( (root1!=null ? root1.right : null), (root2!=null ? root2.right : null) );
        
        return new_node;
    }
}

File: 617-merge-two-binary-trees/617-merge-two-binary-trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* mergeTrees(TreeNode* root1, TreeNode* root2) {
        
        
        if(!root1 and !root2)return nullptr;
        
            
        TreeNode* new_node = new TreeNode();

        if(root1) new_node -> val += root1->val;
        if(root2) new_node -> val += root2->val;
        
        new_node -> left = mergeTrees( (root1 ? root1->left: nullptr) , (root2 ? root2->left : nullptr) ); 
        
        new_node -> right = mergeTrees( (root1 ? root1->right : nullptr), (root2 ? root2->right :nullptr) );

        return new_node;
            
        
        
    }
};